### DISCUSSION TITLE
OeverlayScrollbar dynamically changed transfrom is not visible in DOM element

### DISCUSSION THREAD

**Discussion by mukhammad-akilov** [KingSora/OverlayScrollbars#564] [Q&A] [answered]
**Describe the bug**
OeverlayScrollbar dynamicallychanged transfrom is not visible in DOM changes. For exmaple if I make scroll or auto scroll then **div.os-scrollbar-handle** shuould take styles with transform but I can't see these styles in Google Chrome
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/cd5f0402-087a-4486-8dc4-9961308f40d7)

**To Reproduce**

1. Go to https://kingsora.github.io/OverlayScrollbars/example/overlayscrollbars/ with Chrome browser (or based or browser based on Chrome engine)
2. Open DevTools
3. Scroll down little beet
4. Investiage **div.os-scrollbar-handle** DOM element and see error

**Expected behavior**
In Firefox browser transform property is visible:
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/32996747-f632-4483-83a7-d66e1bcc0013)

**Examples**
For exmaple you can use this default demo:
https://kingsora.github.io/OverlayScrollbars/example/overlayscrollbars/

**Environment**

- Used Operating System(s): Windows 11 (Version 10.0.22621 Build 22621)
- Used Browser(s) (with version):  Google Chrome 116.0.5845.182

The reason I need this property beacuse I created two custom scrollbars and the first one is scrolled automatically that is why I am observing the changes from another scroll container using MutationObserver API and apply transform style to the first one.

**Comment by KingSora** [comment]
Good day @mukhammad-akilov :)

This change was introduced in version 2.3.0. Supported browsers (such as chrome) use the [scrolltimeline api](https://developer.mozilla.org/en-US/docs/Web/API/ScrollTimeline) to move the scrollbar handles. Eventually this will be the way how the scrollbar handles will move in every browser because it improves the performance signifficantly.

Since the value you need is calculated from other values you shouldn't need a MutationObserver at all. Can you post an example of what you exactly wanna create? I would look into it and make an version which works without the MutationObserver.

**Reply by mukhammad-akilov** [reply]
Thank you for detailed reply! The reason why I need this is becauce in our project we have banners where big text scrolled automatically. But when I slowly scroll container by 1px using setInterval or requestAnimatonFrame it does not look smooth at all.
That is why I decided to use transfrom: translateY for this auto scroll.
But I don't scroll container I just apply translateY to its content. 
In our old banners we use custom scrollbar (thanks to your package). 
So to emulate scroll I create another container with same content and make it auto scrollable and hide it from user, then using MutationObserver API I observe scrollbar position of fake container scroll and apply to the visible one

In this image there 4 containers(Red: setInterval, Orange: requestAnimationFrame, green: Transfrom: translateY (the most smooth one), Purple: fake auto scroll container whish won't be visible to user)

That is why I need to observe Dom change of fake scroll container to apply to real one

@KingSora What should I do? Donwgrade to older version of package?

Live demo: https://web-auto-smooth-scroll.vercel.app/
Link to repo: https://github.com/mukhammad-akilov/web-auto-smooth-scroll

![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/a7f37434-6e4f-4f4e-bf4f-38fa4a2c60e4)

**Reply by mukhammad-akilov** [reply]
Also I noticed that in Firefox there 2 scrollbars: default and scrollbar of OverlayScrollbar:
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/0fbea8c7-0c40-44e6-a86a-c16d6fd33e52)

**Reply by KingSora** [reply]
@mukhammad-akilov The scrollbar you mentioned is there because of the `.scroll-boxes-container .scroll-box-container` selector which has the `overflow-y: auto` style and overwrites the `overflow: hidden` style of OverlayScrollbars. The repo Link (https://github.com/mukhammad-akilov/web-auto-smooth-scroll) leads to a 404 page for me... maybe its a private repo?..

**Reply by mukhammad-akilov** [reply]
Thank for you reply :)
It was my mistake, I make the repo public now

**Reply by KingSora** [reply]
@mukhammad-akilov I've made a working solution for you, with the correct calculations for the scrollbar handle: https://stackblitz.com/edit/vitejs-vite-r4pc6h?file=src%2Findex.ts

I hope that this is helpful for you and kinda what you searched for

**Reply by mukhammad-akilov** [reply]
Thanks for you example! I checked it in Firefox and Chrome. In Firefox it is being scrolled, but in Chrome:
Chrome:
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/5c682d39-2049-46c7-a9aa-0066012f504a)

Also I have several questions:
1)In Firefox scrollbar is not scrolled till the end. How can I fix it?
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/7ff6fbd5-8516-4246-b8f1-3d28b3977aaa)

2)How can I decrease and increase the speed of scrollbar scroll? 
For example if I want to increase speed of smooth scroll, I just add more pixels:
![image](https://github.com/KingSora/OverlayScrollbars/assets/31181880/dacae197-b1dc-45ed-92ff-eb70cf50c70b)

But how to sync this speed with custom scrollbar?

Sorry for interupting you and taking your time :(

**Reply by KingSora** [reply, marked as answer]
@mukhammad-akilov I've created a improved example here which hopefully should solve the chrome issue and the "scroll till the end" issue: https://stackblitz.com/edit/vitejs-vite-xiuxru?file=src%252Findex.ts (increased the speed to 100) I've also synced the transition and interval speed to `250ms`. (you can customize this as you want, just keep them in sync to have the best possible experience)

Regarding syncing the speed, what do you exactly mean? You would just need to increase the number

**Reply by mukhammad-akilov** [reply]
Thank you very much! You saved my time :) For now everything works fine. I will learn deeper your example. By syncing the speed I ment about scrolling content and moving scrollbar, but for now everything works fine. Thanks a lot!

**Reply by KingSora** [reply]
Youre welcome :)

Please click the "Mark as anwer" button on my reply which helped you the post to mark the discussion as resolved
